src/lib/parse_format.rs

```diff
@@ -20,45 +10,69 @@ pub enum FormatToken {
     Literal(String),
 }
 
-// This newtype is needed to sidestep StructOpt's Vec behaviour
-#[derive(Debug)]
-pub struct Format(pub Vec<FormatToken>);
+// Get around structopt automatic Vec handling.
+pub(crate) type Format = Vec<FormatToken>;
+
+pub(crate) fn parse_format_string(input: &str) -> Result<Format, String> {
+    let mut tokens = Vec::new();
+    let mut input = input.as_bytes();
 
-pub fn parse_format_string(input: &str) -> Result<Format, String> {
-    match parse_all(input) {
-        Ok(("", v)) => Ok(Format(v)),
-        Err(s) => Err(format!("Format string parse error: {:?}", s)),
-        Ok((s, _)) => Err(format!("Format string parse error near \"{}\"", s)),
+    loop {
```

Comment:
It would be nice to replace this with a different loop construct, but I'm not sure what.

Reply:
Yeah, but I had not found a better way.